[memory profiling] show pre-existing memory in trace_plot

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97590
* #97406

Previously we only plotted memory if it was allocated or freed while
trace recording was active. This change also adds any pre-existing blocks
to the visualization. This helps because it is common to enable trace recording
later and then not realize that there is a lot of allocated memory in
the trace eventhough a lot was allocated beforehad.

zdevito committed:
[memory profiling] show pre-existing memory in trace_plot

Previously we only plotted memory if it was allocated or freed while
trace recording was active. This change also adds any pre-existing blocks
to the visualization. This helps because it is common to enable trace recording
later and then not realize that there is a lot of allocated memory in
the trace eventhough a lot was allocated beforehad.

[ghstack-poisoned]

zdevito commented on torch/cuda/_memory_viz.py:
```diff
@@ -1359,7 +1376,7 @@ class Event {
             .append('rect')
             .attr('x', (x) => xScale(x.segment.offset + (x.addr - x.segment.addr)))
             .attr('y', (x) => yScale(x.segment.row))
-            .attr('width', (x) => xScale(x.real_size > padding/4 ? x.real_size - padding/4 : x.real_size))
```
this padding looks too much like free space

zdevito committed:
Update on "[memory profiling] show pre-existing memory in trace_plot"


Previously we only plotted memory if it was allocated or freed while
trace recording was active. This change also adds any pre-existing blocks
to the visualization. This helps because it is common to enable trace recording
later and then not realize that there is a lot of allocated memory in
the trace eventhough a lot was allocated beforehad.

[ghstack-poisoned]

eellison reviewed (commented):
Nice !! Would you mind posting a picture of the change?

zdevito committed:
Update on "[memory profiling] show pre-existing memory in trace_plot"


Previously we only plotted memory if it was allocated or freed while
trace recording was active. This change also adds any pre-existing blocks
to the visualization. This helps because it is common to enable trace recording
later and then not realize that there is a lot of allocated memory in
the trace eventhough a lot was allocated beforehad.

[ghstack-poisoned]

zdevito commented:
It looks similar to before, but now at the bottom is any memory that already exists at the beginning of recording.
![Screenshot 2023-03-27 at 9 35 08 PM](https://user-images.githubusercontent.com/370202/228129230-a3780ed2-2b21-45df-9dd3-4ea27e009861.png)

eellison reviewed (approved):
cool!
